apollo-integration/src/test/java/com/apollographql/android/impl/ResponseNormalizationTest.java

```diff
@@ -0,0 +1,348 @@
+package com.apollographql.android.impl;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+
+import com.apollographql.android.ApolloCall;
+import com.apollographql.android.CustomTypeAdapter;
+import com.apollographql.android.api.graphql.Response;
+import com.apollographql.android.cache.normalized.ApolloRecord;
+import com.apollographql.android.cache.normalized.CacheKeyResolver;
+import com.apollographql.android.cache.normalized.CacheReference;
+import com.apollographql.android.cache.normalized.InMemoryNormalizedCacheStore;
+import com.apollographql.android.impl.normalizer.EpisodeHeroName;
+import com.apollographql.android.impl.normalizer.HeroName;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import okhttp3.OkHttpClient;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static com.apollographql.android.impl.normalizer.type.Episode.JEDI;
+import static com.google.common.truth.Truth.assertThat;
+
+public class ResponseNormalizationTest {
+
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
+
+  ApolloClient apolloClient;
+  MockWebServer server;
+  InMemoryNormalizedCacheStore cacheStore;
+
+  @Before public void setUp() {
+    server = new MockWebServer();
+    CustomTypeAdapter<Date> dateCustomTypeAdapter = new CustomTypeAdapter<Date>() {
+      @Override public Date decode(String value) {
+        try {
+          return DATE_FORMAT.parse(value);
+        } catch (ParseException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Override public String encode(Date value) {
+        return DATE_FORMAT.format(value);
+      }
+    };
+
+    OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+    cacheStore = new InMemoryNormalizedCacheStore();
+
+    apolloClient = ApolloClient.builder()
+        .serverUrl(server.url("/"))
+        .okHttpClient(okHttpClient)
+        .normalizedCache(cacheStore, new CacheKeyResolver() {
+          @Nullable @Override public String cacheKeyForJsonObject(Map<String, Object> jsonObject) {
```

Comment:
nitpick: method should be named resolve unless there are other variants of how the resolver can work